tests/integration/renderer/test_persistence.py

```diff
@@ -526,3 +532,189 @@ def set_out(val):
+    app.layout = html.Div(
+        [
+            html.Button("fire callback", id="btn"),
+            html.Div(
+                children=[
+                    dcc.DatePickerRange(
+                        id="dpr1",
+                        start_date=datetime.today() - td(days=3),
+                        end_date=datetime.today(),
+                        persistence=True,
+                        persistence_type="session",
+                    ),
+                    html.P("dpr1", id="dpr1-p-start"),
+                    html.P("dpr1", id="dpr1-p-end"),
+                    html.Div(id="container"),
+                    html.P("dpr2", id="dpr2-p-start"),
+                    html.P("dpr2", id="dpr2-p-end"),
+                ]
+            ),
+        ]
+    )
+
+    @app.callback(Output("container", "children"), [Input("btn", "n_clicks")])
+    def update_output(value):
+        return dcc.DatePickerRange(
+            id="dpr2",
+            start_date=datetime.today() - td(days=3),
+            end_date=datetime.today(),
+            persistence=True,
+            persistence_type="session",
+        )
+
+    @app.callback(Output("dpr1-p-start", "children"), [Input("dpr1", "start_date")])
+    def display_dps1(value):
+        print(value)
+        return value
+
+    @app.callback(Output("dpr1-p-end", "children"), [Input("dpr1", "end_date")])
+    def display_dps1(value):
+        print(value)
+        return value
+
+    @app.callback(Output("dpr2-p-start", "children"), [Input("dpr2", "start_date")])
+    def display_dps2(value):
+        return value
+
+    @app.callback(Output("dpr2-p-end", "children"), [Input("dpr2", "end_date")])
+    def display_dps2(value):
+        return value
+
+    dash_duo.start_server(app)
+    dpr1 = dash_duo.find_element("div#dpr1 div div div div .DateInput_input")
+    dpr2 = dash_duo.find_element("div#dpr2 div div div div .DateInput_input")
+
+    (
+        ActionChains(dash_duo.driver)
+        .move_to_element(dpr1)
+        .click(dpr1)
+        .send_keys(Keys.END)
+        .key_down(Keys.SHIFT)
+        .send_keys(Keys.HOME)
+        .key_up(Keys.SHIFT)
+        .send_keys("01/01/2020")
+        .pause(0.2)
+        .send_keys(Keys.END)
+        .key_down(Keys.SHIFT)
+        .send_keys(Keys.HOME)
+        .key_up(Keys.SHIFT)
+        .send_keys("01/02/2020")
+    ).perform()
+
+    dash_duo.wait_for_text_to_equal("#dpr1-p-start", "2020-01-01")
+    dash_duo.wait_for_text_to_equal("#dpr1-p-end", "2020-01-02")
+
+    (
+        ActionChains(dash_duo.driver)
+        .move_to_element(dpr2)
+        .click(dpr2)
+        .send_keys(Keys.END)
+        .key_down(Keys.SHIFT)
+        .send_keys(Keys.HOME)
+        .key_up(Keys.SHIFT)
+        .send_keys("01/01/2020")
+        .pause(0.2)
+        # .send_keys("01/02/2020")
+        .send_keys(Keys.END)
+        .key_down(Keys.SHIFT)
+        .send_keys(Keys.HOME)
+        .key_up(Keys.SHIFT)
+        .send_keys("01/02/2020")
```

Comment:
DRY 🌴🐫

Reply:
fe58b2a

Reply:
There's also `dash_duo.clear_input` that would simplify some of these (assuming there isn't some reason it fails for datepickers)

https://github.com/plotly/dash/blob/fe58b2a697bbb185fd58fd2d413f5022e66afb6f/tests/integration/renderer/test_persistence.py#L401